src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -51,18 +52,26 @@ internal abstract class AbstractSnippetInfoService : ForegroundThreadAffinitized
             IAsynchronousOperationListenerProvider listenerProvider)
             : base(threadingContext)
         {
-            AssertIsForeground();
-
             if (serviceProvider != null)
             {
-                var textManager = (IVsTextManager2)serviceProvider.GetService(typeof(SVsTextManager));
-                if (textManager.GetExpansionManager(out _expansionManager) == VSConstants.S_OK)
-                {
-                    ComEventSink.Advise<IVsExpansionEvents>(_expansionManager, this);
-                    _waiter = listenerProvider.GetListener(FeatureAttribute.Snippets);
-                    _languageGuidForSnippets = languageGuidForSnippets;
-                    PopulateSnippetCaches();
-                }
+                _waiter = listenerProvider.GetListener(FeatureAttribute.Snippets);
+                _languageGuidForSnippets = languageGuidForSnippets;
+
+                InitializeAndPopulateSnippetsCacheAsync(threadingContext, serviceProvider).Forget();
```

Comment:
Did you mean to Task.Run() this? Otherwise the first bit of InitializeAndPopulateSnippetsCacheAsync may still be running on your UI thread. (It's not really problematic here, but...)

Reply:
`InitializeAndPopulateSnippetsCacheAsync` immediately yields, so there is no need to do so here.